core/src/sql/fetch.rs

```diff
@@ -39,22 +44,68 @@ impl InfoStructure for Fetchs {
 	}
 }
 
-#[revisioned(revision = 1)]
+#[revisioned(revision = 2)]
 #[derive(Clone, Debug, Default, Eq, PartialEq, PartialOrd, Serialize, Deserialize, Hash)]
 #[cfg_attr(feature = "arbitrary", derive(arbitrary::Arbitrary))]
 #[non_exhaustive]
-pub struct Fetch(pub Value);
+pub struct Fetch(
+	#[revision(end = 2, convert_fn = "convert_fetch_idiom")] pub Idiom,
+	#[revision(start = 2)] pub Value,
+);
+
+impl Fetch {
+	fn convert_fetch_idiom(&mut self, _revision: u16, old: Idiom) -> Result<(), revision::Error> {
+		self.1 = Value::Idiom(old);
+		Ok(())
+	}
+
+	pub(crate) async fn compute(
+		&self,
+		stk: &mut Stk,
+		ctx: &Context<'_>,
+		opt: &Options,
+	) -> Result<Cow<Idiom>, Error> {
+		let i = match &self.1 {
+			Value::Idiom(idiom) => Cow::Borrowed(idiom),
+			Value::Param(param) => {
+				let p = param.compute(stk, ctx, opt, None).await?;
+				match p {
+					Value::Strand(s) => Cow::Owned(Idiom::from(s.0)),
+					Value::Idiom(i) => Cow::Owned(i),
```

Comment:
How can you return an idiom from a parameter?

Reply:
Well, the compute method returns a Value, I am not sure there is any contract the prevent this method to return an idiom.

We could imagine that this would work:

```sql
let $param = purchase.out;
```